packages/cafs/src/writeFile.ts

```diff
@@ -0,0 +1,20 @@
+import makeDir = require('make-dir')
+import fs = require('mz/fs')
+import path = require('path')
+
+const dirs = new Set()
+
+// write a stream to destination file
+export default async function (
```

Comment:
I believe this can throw, but there is no try/catch.

Reply:
Well, the target file will never exist, so it won't fail. If the reason of failure is something else, then it should be fine to allow the exception to stop the installation.

Reply:
Fine, but I think the installation shouldn't stop with an ugly error message from node, and in the future, instant process termination.

Reply:
It will throw from here

https://github.com/pnpm/pnpm/blob/f385c82cf7cfdcfbc52a7ad9d73d5424910f6082/packages/supi/src/install/link.ts#L251

And that will be catched by the global try/catch that is in the CLI package

Reply:
Хорошо. Does anyone other than you have a good grasp of the source code? I think it would take me weeks to get a thorough understanding of how it all fits together.

Reply:
There are some people that know different parts of the source code. Of course, I know the most as I did most of the changes. Unfortunately, there are many parts of the code that are very ugly and hard to understand. If you'll have questions, I can answer in the Gitter chatroom.